### PR TITLE
Extend ADS Sensor documentation

### PR DESCRIPTION
**Pull request by mrpasztoradam**
## Proposed change
Extend ADS Sensor documentation

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/125331
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Expanded the documentation for the ADS sensor platform to include additional supported data types: BOOL, SINT, USINT, WORD, DWORD, REAL, and LREAL.
  
- **Documentation**
	- Updated the description of the `adstype` configuration option to reflect the newly supported data types, enhancing clarity and usability for developers.

### CODE REVIEW

File: source/_integrations/ads.markdown [hunk 1]
```diff
@@ -27,6 +27,8 @@ The ADS (automation device specification) describes a device-independent and fie
 
 There is currently support for the following device types within Home Assistant:
 
+- [Configuration](#configuration)
+- [Action](#action)
```
Comment (klaasnicolaas) [inline source/_integrations/ads.markdown]: This doesn't belong on the list

**Review by klaasnicolaas — DISMISSED** [review]

File: source/_integrations/ads.markdown (hunk 1 above)
Comment (mrpasztoradam) [reply source/_integrations/ads.markdown]: Its automatically added to the list when saving. Please advise.

Comment (mrpasztoradam) [reply source/_integrations/ads.markdown]: fixed

**Review by frenck — APPROVED** [review]
Thanks, @mrpasztoradam 👍 

../Frenck

**Pull request merged**